src/ImageProcessor.Web/Caching/DiskCache.cs

```diff
@@ -255,15 +255,19 @@ public override Task TrimCacheAsync()
                                     ImageProcessorBootstrapper.Instance.Logger.Log<DiskCache>($"Unable to clean cached file: {fileInfo.FullName}, {ex.Message}");
                                 }
                             }
+
+                            // If the directory is empty, delete it to remove the FCN
+                            if (Directory.GetFiles(directory, "*", SearchOption.TopDirectoryOnly).Length == 0
```

Comment:
We already get all files in the current directory above and keep track of the total count (and decrement it when deleting a file), so this can be simplified to:
```suggestion
                            if (count == 0
```